rollup/sync_service.go

```diff
@@ -859,13 +888,20 @@ func (s *SyncService) ProcessTransactionEnqueuedLog(ctx context.Context, ethlog
 	// Nonce is set by god key at execution time
 	// Value and gasPrice are set to 0
 	tx := types.NewTransaction(uint64(0), event.Target, big.NewInt(0), event.GasLimit.Uint64(), big.NewInt(0), event.Data, &event.L1TxOrigin, new(big.Int).SetUint64(ethlog.BlockNumber), types.QueueOriginL1ToL2, types.SighashEIP155)
-	// Timestamp is used to update the blockchains clocktime
-	timestamp := time.Unix(event.Timestamp.Int64(), 0)
-	rtx := RollupTransaction{tx: tx, timestamp: timestamp, blockHeight: ethlog.BlockNumber, executed: false, index: event.QueueIndex.Uint64()}
+	// Set the timestamp in the txmeta
+	timestamp := uint64(event.Timestamp.Int64())
+	tx.SetL1Timestamp(timestamp)
+
+	rtx := RollupTransaction{tx: tx, blockHeight: ethlog.BlockNumber, executed: false, index: event.QueueIndex.Uint64()}
 	// In the case of a reorg, the rtx at a certain index can be overwritten
 	s.txCache.Store(event.QueueIndex.Uint64(), &rtx)
 	log.Debug("Transaction enqueued", "queue-index", event.QueueIndex.Uint64(), "timestamp", timestamp, "l1-blocknumber", ethlog.BlockNumber, "to", event.Target.Hex())
 
+	// Set the timestamp and blocknumber so that transactions from
+	// queue origin sequencer can access this information
```

Comment:
Are there any other times that the latest L1 timestamp/blocknumber need to be updated?